### PR TITLE
#1790 - SMS opt-in to system imports with mobile numbers

### LINKED ISSUE #1790: add SMS opt-in to system imports with mobile numbers
Part of epic #1017, related to #1788 and #1789

AS an **admin**,
I WANT to see an SMS opt-in checkbox when I attempt to import users with mobile numbers,
SO THAT I can confirm the mobile numbers I'm importing are opted in to SMS notifications.

### How it works now
There is currently no opt-in confirmation required before importing any data onto the platform. 

### How it should work instead 
After an admin clicks "import volunteers csv" or "import supervisors csv", they should see a dialogue box _if_ their import contains mobile numbers (if the mobile number column is not empty). The dialogue box should contain a checkbox with the following prompt:

_Please check this box to verify that these mobile numbers have opted in to receive SMS notifications. (They will have the opportunity to opt out or update their preferences.)_

After checking the box, the import resumes. The import _can not_ resume without checking the box.

### PR DESCRIPTION
**Pull request by harsohailB**
### What github issue is this PR for, if any?
Resolves #1790

### What changed, and why?

- [x] Write function to detect mobile numbers exist
- [x] Create opt-in modal
- [x] Pause import if opt-in required and show modal
- [x] Temporarily store CSV in local storage
- [x] Delete CSV from local storage upon successful import
- [x] Resume import upon confirming opt-in 
- [x] Turn on SMS notifications for imported users

### How will this affect user permissions?
- Volunteer permissions: N/A
- Supervisor permissions: N/A
- Admin permissions: N/A

### How is this tested? (please write tests!) 💖💪

- [x] Update import request tests
- [x] Update volunteer and supervisor importer tests
- [x] Add system tests for opt in modal

### Screenshots please :)

image

### CODE REVIEW

File: app/controllers/imports_controller.rb [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+  require "csv"
+
   include ActionView::Helpers::UrlHelper
   after_action :verify_authorized
 
   def index
     authorize :import
     @import_type = params.fetch(:import_type, "volunteer")
     @import_error = session[:import_error]
+    @sms_opt_in_error = session[:sms_opt_in_error]
```
Comment (xihai01) [inline app/controllers/imports_controller.rb:11]: nit: the modal doesn't really represent an error. It's more like a warning or caution for the user.

**Review by xihai01 — COMMENTED** [review]
lgtm

File: app/controllers/imports_controller.rb (hunk 1 above)
Comment (harsohailB) [reply app/controllers/imports_controller.rb]: Renamed as a warning!

**Review by xihai01 — APPROVED** [review]
wow awesome work. I think it looks good. Maybe the others will want to take a look too?

**Comment by 7riumph** [comment]
> wow awesome work. I think it looks good. Maybe the others will want to take a look too?

I agree, it looks great to me, though @Compwron /a maintainer should definitely have a look since we’ve reached into javascript territory with the CSV file caching additions in `app/javascript/src/import.js`

**Review by 7riumph — APPROVED** [review]

**Review by compwron — APPROVED** [review]

**Comment by compwron** [comment]
very very cool!!!

**Pull request merged**